proposals/NNNN-oracle-nio.md

```diff
@@ -0,0 +1,109 @@
+`OracleConnection` instanciates a single connection to the database, whereas `OracleClient` creates a pool of connections.
+Both take a [`OracleConnection.Configuration`](https://swiftpackageindex.com/lovetodream/oracle-nio/main/documentation/oraclenio/oracleconnection/configuration) parameter to configure the individual connections.
+
+```swift
+let logger = Logger(label: "my-oracle-logger")
+
+let config = OracleConnection.Configuration(
+    host: "127.0.0.1", 
+    port: 1521,
+    service: .serviceName("my_service"),
+    username: "my_username",
+    password: "my_password"
+)
+
+let connection = try await OracleConnection.connect(
+  configuration: config,
+  id: 1,
+  logger: logger
+)
+```
+
+With the now successfully established connection, one can start to query the database using the [`query(_:)`](https://swiftpackageindex.com/lovetodream/oracle-nio/main/documentation/oraclenio/oracleconnection/query(_:options:logger:file:line:)) method. 
+This method allows the user to run a Query, DML, DDL or [PL/SQL](https://www.oracle.com/database/technologies/appdev/plsql.html).
+
+```swift
+let rows = try await connection.query("SELECT id, username, birthday FROM users")
+```
+
+A query returns an [`OracleRowSequence`](https://swiftpackageindex.com/lovetodream/oracle-nio/main/documentation/oraclenio/oraclerowsequence), which is an AsyncSequence of [`OracleRow`](https://swiftpackageindex.com/lovetodream/oracle-nio/main/documentation/oraclenio/oraclerow)s. The rows can either be collected altogether or iterated one-by-one.
+It's possible to decode rows to a set of Swift types directly. 
+
+```swift
+for try await (id, username, birthday) in rows.decode((Int, String, Date).self) {
+  // do something with the datatypes.
+}
+```
+
+A type must implement the [`OracleDecodable`](https://swiftpackageindex.com/lovetodream/oracle-nio/main/documentation/oraclenio/oracledecodable) protocol in order to be decoded from a row. 
+OracleNIO provides default implementations for most of Swift's builtin types, as well as some types provided by Foundation. 
+Users of the library can implement their own `OracleDecodable` types, too.
+
+Once done, the database has to be closed.
+
+```swift
+try await connection.close()
+```
+
+An up-to-date documentation of the public API can be found on the [Swift Package Index](https://swiftpackageindex.com/lovetodream/oracle-nio/main/documentation/oraclenio).
+
+It should be noted that the public API and many of the internals are very similar to [PostgresNIO](https://github.com/vapor/postgres-nio), which was the main inspiration in the process of creating OracleNIO. This should make adoption of OracleNIO easier to those already familiar with PostgresNIO.
```

Comment:
There is an opportunity here to standardize the interface between the two project and create a JDBC like for relational database access in Swift ? This is out of the scope of this project, just thinking big.

Reply:
That'd be great!